src/OSPSuite.R/Services/ConcurrentSimulationRunner.cs

```diff
@@ -0,0 +1,74 @@
+﻿using OSPSuite.Core.Domain;
+using OSPSuite.Core.Domain.Data;
+using OSPSuite.Core.Domain.Services;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using SimulationRunOptions = OSPSuite.R.Domain.SimulationRunOptions;
+
+namespace OSPSuite.R.Services
+{
+   public interface IConcurrentSimulationRunner
+   {
+      /// <summary>
+      /// General simulation options
+      /// </summary>
+      SimulationRunOptions SimulationRunOptions { get; set; }
+
+      /// <summary>
+      /// Number of cores to use concurrently. Use 0 or a negative value for using the maximum available.
+      /// </summary>
+      int NumberOfCores { get; set; }
+
+      /// <summary>
+      /// List of Simulations
+      /// </summary>
+      void AddSimulation(IModelCoreSimulation simulation);
```

Comment:
reason why you are not passing the list of simulations in RunConcurrently?

Reply:
So you can prepare everything once combining IModelCoreSimulation objects and SimulationBatch objects. Even if you pass the lists (two instead of one) from R you will need to have Add functions on the settings object you will provide. Remember you cannot pass a list of objects directly from R. I think this way, having the wrapper object handling the settings through functions is more compact for the R caller.

Reply:
You can absolutely pass array to R using args object as we discussed I believe. and if not, please see here
https://github.com/Open-Systems-Pharmacology/PK-Sim/blob/develop/src/PKSim.R/Domain/IndividualCharacteristics.cs#L9

The problem with the implementation is that the caller needs to remember to clear. I find side effect-free stuff much easier to deal with in general. That's why the SimulationRunner takes the simulation as parameter. I am not sure why the ConcurrentSImulationRunner should have a different logic

Reply:
But let's see what the interface in R will be to be honest. This should drive the .NET implementation

Reply:
What I meant by passing the list directly is that you have to create a settings object with add methods... so it means a bit more of wrapping. Maybe I am wrong and @PavelBal could help here but I think that you would normally prepare the settings once and use it several times so I do not see a need for clearing in between. Let us say you are in a PI scenario, you will need to prepare your simulations and then invoke the RunConcurrently again and again with the same simulations just modifying the values you want to set as inputs but the set of simulations should not change since your scenario has not changed at all. The way we have set the api now you would prepare your ConcurrentSimulationRunner once and use it once and again in every iteration by just specifying the input values to set to the simulation objects by directly setting this if use case 1 or by the use of the appropriate SimulationBatchRunValues.

Reply:
oh I see. Fair enough